In [11]:
import pandas as pd 
import numpy as np 
import matplotlib.pyplot as plt 
import seaborn as sns 

from pyfinmod.wacc import wacc, cost_of_debt, cost_of_equity
from pyfinmod.financials import Financials

import tabula

# 1 - What is the WACC for Marriott?

$$
    WACC = (1-\tau) * Cost_{debt} * (\frac{D}{D+E}) + Cost_{equity} * (\frac{E}{D+E})
$$

### 1987 Financial Summary of Marriott

| Total Assets | Long-term Debt | Equity |
| ------------ | -------------- | ------ |
| 5370.5 | 2498.8 | 2871.7 |


In [6]:
E = 2871.7
D = 2498.8
V = E+D 

print(f"Value is {V}, Equity is {E}, and Debt is {D}. \nThe percent of Debt to Value is {D/V}, and the ratio of Debt to Equity is {D/E}")

Value is 5370.5, Equity is 2871.7, and Debt is 2498.8. 
The percent of Debt to Value is 0.4652825621450517, and the ratio of Debt to Equity is 0.8701466030574225


In [8]:
tax_rate = 175.9/398.9
d_to_v = D/V 
d_to_e = D/E 

print(tax_rate)

0.4409626472800201


In [14]:
tabula.read_pdf("S2.1_ACF_Marriott Corporation.pdf", pages=6)[0]

Unnamed: 0     1978     1979     1980     1981  \
0                   Summary of Operations      NaN      NaN      NaN      NaN   
1                                   Sales  1,174.1  1,426.0  1,633.9  1,905.7   
2        Earnings before interest expense      NaN      NaN      NaN      NaN   
3                        and income taxes    107.1    133.5    150.3    173.3   
4                        Interest expense     23.7     27.8     46.8     52.0   
5              Income before income taxes     83.5    105.6    103.5    121.3   
6                            Income taxes     35.4     43.8     40.6     45.2   
7      Income from continuing operationsa     48.1     61.8     62.9     76.1   
8                              Net income     54.3     71.0     72.0     86.1   
9   Funds provided from cont. operationsb    101.2    117.5    125.8    160.8   
10             Capitalization and Returns      NaN      NaN      NaN      NaN   
11                           Total assets  1,000.3  1,080.4  1,214.3  1,454.9   
12                         Total capitalc    826.9    891.9    977.7  1,167.5   
13                         Long-term debt    309.9    365.3    536.6    607.7   
14                Percent to total capita    37.5%    41.0%    54.9%    52.1%   
15                   Shareholders’ equity    418.7    413.5    311.5    421.7   
16               Per Share and Other Data      NaN      NaN      NaN      NaN   
17                    Earnings per share:      NaN      NaN      NaN      NaN   
18                 aContinuing operations      .25      .34      .45      .57   
19                             Net income      .29      .39      .52      .64   
20                         Cash dividends     .026     .034     .042     .051   
21                   Shareholders’ equity     2.28     2.58     2.49     3.22   
22               Market price at year end     2.43     3.48     6.35     7.18   
23       Shares outstanding (in millions)    183.6    160.5    125.3    130.8   
24    Return on avg. shareholders’ equity    13.9%    17.0%    23.8%    23.4%   

       1982     1983     1984     1985     1986     1987  
0       NaN      NaN      NaN      NaN      NaN      NaN  
1   2,458.9  2,950.5  3,524.9  4,241.7  5,266.5  6,522.2  
2       NaN      NaN      NaN      NaN      NaN      NaN  
3     205.5    247.9    297.7    371.3    420.5    489.4  
4      71.8     62.8     61.6     75.6     60.3     90.5  
5     133.7    185.1    236.1    295.7    360.2    398.9  
6      50.2     76.7    100.8    128.3    168.5    175.9  
7      83.5    108.4    135.3    167.4    191.7    223.0  
8      94.3    115.2    139.8    167.4    191.7    223.0  
9     203.6    272.7    322.5    372.3    430.3    472.8  
10      NaN      NaN      NaN      NaN      NaN      NaN  
11  2,062.6  2,501.4  2,904.7  3,663.8  4,579.3  5,370.5  
12  1,634.5  2.007.5  2,330.7  2,861.4  3,561.8  4,247.8  
13    889.3  1,071.6  1,115.3  1,192.3  1,662.8  2,498.8  
14    54.4%    53.4%    47.9%    41.7%    46.7%    58.8%  
15    516.0    628.2    675.6    848.5    991.0    810.8  
16      NaN      NaN      NaN      NaN      NaN      NaN  
17      NaN      NaN      NaN      NaN      NaN      NaN  
18      .61      .78     1.00     1.24     1.40     1.67  
19      .69      .83     1.04     1.24     1.40     1.67  
20     .063     .076     .093     .113     .136      .17  
21     3.89     4.67     5.25     6.48     7.59     6.82  
22    11.70    14.25    14.70    21.58    29.75    30.00  
23    132.8    134.4    128.8    131.0    130.6    118.8  
24    20.0%    20.0%    22.1%    22.1%    20.6%    22.2%

In [21]:
# equity beta of Marriott is 1.11
beta_unl = 1.11 / (1 + d_to_e)
mar_cost_of_debt = 0.0872 + 0.013

print(beta_unl, mar_cost_of_debt)

0.5935363560189927 0.1002


In [135]:
def betaLeverCalc(beta_unl, debt_to_equity=0, debt_to_value=0):

    if debt_to_value > 0:
        debt_to_equity = debt_to_value / (1-debt_to_value)
        
        return round(beta_unl * (1+debt_to_equity), 5)

    if debt_to_equity > 0:

        return round(beta_unl * (1+debt_to_equity), 5)

In [69]:
betaLeverCalc(0.5935, debt_to_value=0.6)

1.48375

In [70]:
# relever the beta with the new target
tgt_d_to_e = 1.5 

tgt_beta_e = beta_unl * (1+tgt_d_to_e)

In [30]:
print(f"New target equity beta is {tgt_beta_e}")

New target equity beta is 1.4838408900474818


In [36]:
# cost of equity 

cost_of_equity(tgt_beta_e, 0.0872, 0.0743+0.0872)

0.1974493781305279

In [43]:
wacc

<function pyfinmod.wacc.wacc(equity, balance_sheet, income_statement, beta, risk_free_interest_rate, market_return)>

In [44]:
cost_of_equity(tgt_beta_e, 0.0872, 0.0743+0.0872)*0.4 + mar_cost_of_debt*(1-tax_rate)*0.6

0.11258907689773635


### WACC for Marriott
11.26%

# 2 - What is the problem with using WACC of Marriott for evaluating projects in different divisions?

# 3 - WACC for Lodging, Restaurant, and Contract Services

In [53]:
tabula.read_pdf("S2.1_ACF_Marriott Corporation.pdf", pages=7, pandas_options={"header":None})[0]

0      1        2        3        4        5        6
0       Operating profit  132.6    139.7    161.2    185.8    215.7    263.9
1    Identifiable assets  909.7  1,264.6  1,786.3  2,108.9  2,236.7  2,777.4
2           Depreciation   22.7     27.4     31.3     32.4     37.1     43.9
3   Capital expenditures  371.5    377.2    366.4    808.3    966.6  1,241.9
4     Contract Services:    NaN      NaN      NaN      NaN      NaN      NaN
5                  Sales  819.8    950.6  1,111.3  1,586.3  2,236.1  2,969.0
6       Operating profit   51.0     71.1     86.8    118.6    154.9    170.6
7    Identifiable assets  373.3    391.6    403.9    624.4  1,070.2  1,237.7
8           Depreciation   22.9     26.1     28.9     40.2     61.1     75.3
9   Capital expenditures  127.7     43.8     55.6    125.9    448.7    112.7
10          Restaurants:    NaN      NaN      NaN      NaN      NaN      NaN
11                 Sales  547.4    679.4    707.0    757.0    797.3    879.9
12      Operating profit   48.5     63.8     79.7     78.2     79.1     82.4
13   Identifiable assets  452.2    483.0    496.7    582.6    562.3    567.6
14          Depreciation   25.1     31.8     35.5     34.8     38.1     42.1
15  Capital expenditures  199.6     65.0     72.3    128.4     64.0     79.6

In [54]:
lodging = {
    "id_assets": 2777.4
}

contract_services = {
    "id_assets": 1237.7
}

restaurants = {
    "id_assets": 567.6
}

In [74]:
tabula.read_pdf("S2.1_ACF_Marriott Corporation.pdf", pages=8, pandas_options={"header":None})[0].dropna()

0      1     2    3     4
3          MARRIOTT CORPORATION  22.4%  1.11  41%  6.52
7     HILTON HOTELS CORPORATION   13.3   .76  14%  0.77
9           HOLIDAY CORPORATION   28.8  1.35  79%  1.66
12         LA QUINTA MOTOR INNS   -6.4   .89  69%  0.17
14            RAMADA INNS, INC.   11.7  1.36  65%  0.75
17       CHURCH’S FRIED CHICKEN   -3.2  1.45   4%  0.39
20  COLLINS FOODS INTERNATIONAL   20.3  1.45  10%  0.57
23         FRISCH’S RESTAURANTS   56.9   .57   6%  0.14
25            LUBY’S CAFETERIAS   15.1   .76   1%  0.23
27                   McDONALD’S   22.5   .94  23%  4.89
29        WENDY’S INTERNATIONAL    4.6  1.32  21%  1.05

In [123]:
df = tabula.read_pdf("S2.1_ACF_Marriott Corporation.pdf", pages=8, pandas_options={"header":None})[0].dropna()


In [83]:
hotels = df.iloc[[1,2,3,4],:]
restaurants = hotesl = df.iloc[[5,6,7,8,9, 10],:]

In [84]:
hotels

0     1     2    3     4
7   HILTON HOTELS CORPORATION  13.3   .76  14%  0.77
9         HOLIDAY CORPORATION  28.8  1.35  79%  1.66
12       LA QUINTA MOTOR INNS  -6.4   .89  69%  0.17
14          RAMADA INNS, INC.  11.7  1.36  65%  0.75

In [85]:
restaurants

0     1     2    3     4
17       CHURCH’S FRIED CHICKEN  -3.2  1.45   4%  0.39
20  COLLINS FOODS INTERNATIONAL  20.3  1.45  10%  0.57
23         FRISCH’S RESTAURANTS  56.9   .57   6%  0.14
25            LUBY’S CAFETERIAS  15.1   .76   1%  0.23
27                   McDONALD’S  22.5   .94  23%  4.89
29        WENDY’S INTERNATIONAL   4.6  1.32  21%  1.05

In [105]:
hotels.rename(columns={2:"equity_beta"}, inplace=True)

/home/xuren/miniconda3/lib/python3.7/site-packages/pandas/core/frame.py:4238: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  return super().rename(**kwargs)


In [107]:
hotels.rename(columns={3: "market_leverage"}, inplace=True)

/home/xuren/miniconda3/lib/python3.7/site-packages/pandas/core/frame.py:4238: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  return super().rename(**kwargs)


In [109]:
hotels["market_leverage"] = hotels.apply(lambda row: float(row["market_leverage"].replace("%", ""))/100, axis=1)

/home/xuren/miniconda3/lib/python3.7/site-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """Entry point for launching an IPython kernel.


In [110]:
hotels

0     1 equity_beta  market_leverage     4
7   HILTON HOTELS CORPORATION  13.3         .76             0.14  0.77
9         HOLIDAY CORPORATION  28.8        1.35             0.79  1.66
12       LA QUINTA MOTOR INNS  -6.4         .89             0.69  0.17
14          RAMADA INNS, INC.  11.7        1.36             0.65  0.75

In [113]:
hotels["debt_to_equity"] = hotels.apply(lambda row: row["market_leverage"] / (1-row["market_leverage"]), axis=1)

/home/xuren/miniconda3/lib/python3.7/site-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """Entry point for launching an IPython kernel.


In [114]:
hotels

0     1 equity_beta  market_leverage     4  \
7   HILTON HOTELS CORPORATION  13.3         .76             0.14  0.77   
9         HOLIDAY CORPORATION  28.8        1.35             0.79  1.66   
12       LA QUINTA MOTOR INNS  -6.4         .89             0.69  0.17   
14          RAMADA INNS, INC.  11.7        1.36             0.65  0.75   

    debt_to_equity  
7         0.162791  
9         3.761905  
12        2.225806  
14        1.857143

In [118]:
hotels["unlevered_beta"] = hotels.apply(lambda row: float(row["equity_beta"]) / (row["debt_to_equity"] + 1), axis=1)

/home/xuren/miniconda3/lib/python3.7/site-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """Entry point for launching an IPython kernel.


In [119]:
hotels

0     1 equity_beta  market_leverage     4  \
7   HILTON HOTELS CORPORATION  13.3         .76             0.14  0.77   
9         HOLIDAY CORPORATION  28.8        1.35             0.79  1.66   
12       LA QUINTA MOTOR INNS  -6.4         .89             0.69  0.17   
14          RAMADA INNS, INC.  11.7        1.36             0.65  0.75   

    debt_to_equity  unlevered_beta  
7         0.162791          0.6536  
9         3.761905          0.2835  
12        2.225806          0.2759  
14        1.857143          0.4760

In [120]:
hotels.unlevered_beta.mean()

0.42225

In [126]:
betaLeverCalc(0.42225, debt_to_value=0.74) # levered beta for lodging

1.62404

In [127]:
restaurants.rename(columns={2: "equity_beta", 3: "market_leverage"}, inplace=True)
restaurants["market_leverage"] = restaurants.apply(lambda row: float(row["market_leverage"].replace("%", ""))/100, axis=1)
restaurants["debt_to_equity"] = restaurants.apply(lambda row: row["market_leverage"] / (1-row["market_leverage"]), axis=1)


/home/xuren/miniconda3/lib/python3.7/site-packages/pandas/core/frame.py:4238: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  return super().rename(**kwargs)
/home/xuren/miniconda3/lib/python3.7/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  
/home/xuren/miniconda3/lib/python3.7/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pand

In [130]:
restaurants["unlevered_beta"] = restaurants.apply(lambda row: float(row["equity_beta"]) / (row["debt_to_equity"] + 1), axis=1)

/home/xuren/miniconda3/lib/python3.7/site-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """Entry point for launching an IPython kernel.


In [131]:
restaurants

0     1 equity_beta  market_leverage     4  \
17       CHURCH’S FRIED CHICKEN  -3.2        1.45             0.04  0.39   
20  COLLINS FOODS INTERNATIONAL  20.3        1.45             0.10  0.57   
23         FRISCH’S RESTAURANTS  56.9         .57             0.06  0.14   
25            LUBY’S CAFETERIAS  15.1         .76             0.01  0.23   
27                   McDONALD’S  22.5         .94             0.23  4.89   
29        WENDY’S INTERNATIONAL   4.6        1.32             0.21  1.05   

    debt_to_equity  unlevered_beta  
17        0.041667          1.3920  
20        0.111111          1.3050  
23        0.063830          0.5358  
25        0.010101          0.7524  
27        0.298701          0.7238  
29        0.265823          1.0428

In [132]:
restaurants.unlevered_beta.mean()

0.9586333333333332

In [138]:
betaLeverCalc(0.95863, debt_to_value=0.42) # levered beta for restaurants

1.65281

In [145]:
lod_assets = 2777 	                      
cs_assets = 1238 	                         
res_assets = 568 

In [146]:
tot_assets = lod_assets + cs_assets + res_assets

In [147]:
tot_assets

4583

In [148]:
lod_assets/tot_assets, cs_assets/tot_assets, res_assets/tot_assets

(0.6059349770892428, 0.27012873663539166, 0.12393628627536549)

In [149]:
lod_assets/tot_assets * 0.42 + res_assets/tot_assets * 0.95863

0.37330173248963555

In [150]:
0.59-0.373302

0.21669799999999995

In [151]:
0.2167/0.270129

0.8022093148088506

In [152]:
cs_unlevered_beta = 0.8022

In [154]:
betaLeverCalc(0.8022, debt_to_value=0.4)

1.337

In [155]:
cost_of_equity(1.337, 0.0872, 0.0743+0.0872)

0.1865391